app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp

```diff
@@ -686,16 +692,35 @@ struct DeviceDelegateOculusVR::State {
     // Reorient the headset after controller recenter.
     vrapi_SetPropertyInt(&java, VRAPI_REORIENT_HMD_ON_CONTROLLER_RECENTER, 1);
 
+    const char * appId = OCULUS_6DOF_APP_ID;
+
     const int type = vrapi_GetSystemPropertyInt(&java, VRAPI_SYS_PROP_DEVICE_TYPE);
     if ((type >= VRAPI_DEVICE_TYPE_OCULUSGO_START ) && (type <= VRAPI_DEVICE_TYPE_OCULUSGO_END)) {
       VRB_DEBUG("Detected Oculus Go");
       deviceType = device::OculusGo;
+      appId = OCULUS_3DOF_APP_ID;
     } else if ((type >= VRAPI_DEVICE_TYPE_OCULUSQUEST_START) && (type <= VRAPI_DEVICE_TYPE_OCULUSQUEST_END)) {
       VRB_DEBUG("Detected Oculus Quest");
       deviceType = device::OculusQuest;
+    } else if ((type >= VRAPI_DEVICE_TYPE_GEARVR_START) && (type <= VRAPI_DEVICE_TYPE_GEARVR_END)) {
+      VRB_DEBUG("Detected Gear VR");
+      appId = OCULUS_3DOF_APP_ID;
     } else {
       VRB_DEBUG("Detected Unknown Oculus device");
     }
+
+    ovrRequest result = ovr_PlatformInitializeAndroidAsynchronous(appId, java.ActivityObject, java.Env);
```

Comment:
I wonder if it wouldn't be better to only include the OVR lib and API calls in the store builds?

Reply:
I thought about that but my concern was that if there is a related bug to that Asynchronous load we can detect it earlier on dev builds

Reply:
But I thought you said it was failing already in local builds? I must be confused.